pydantic/main.py

```diff
@@ -110,26 +108,26 @@ def hash_func(self_: Any) -> int:
 
             cls.__pydantic_decorators__ = _decorators.gather_decorator_functions(cls)
 
-            # FIXME all generics related attributes should be moved into a dict, like `__pydantic_decorators__`
-            parent_typevars_map = {}
-            for base in bases:
-                base_typevars_map = getattr(base, '__pydantic_generic_typevars_map__', None)
-                if base_typevars_map:
-                    parent_typevars_map.update(base_typevars_map)
-
-            cls.__pydantic_generic_args__ = __pydantic_generic_args__
-            cls.__pydantic_generic_origin__ = __pydantic_generic_origin__
-            cls.__pydantic_generic_parameters__ = __pydantic_generic_parameters__ or getattr(
-                cls, '__parameters__', None
-            )
-            cls.__pydantic_generic_defaults__ = None if not cls.__pydantic_generic_parameters__ else {}
-            if __pydantic_generic_origin__ is None:
-                cls.__pydantic_generic_typevars_map__ = None
+            # Use the getattr below to grab the __parameters__ from the `typing.Generic` parent class
+            if __pydantic_generic_metadata__:
+                cls.__pydantic_generic_metadata__ = __pydantic_generic_metadata__
             else:
-                new_typevars_map = dict(
-                    zip(_generics.iter_contained_typevars(__pydantic_generic_origin__), __pydantic_generic_args__ or ())
-                )
-                cls.__pydantic_generic_typevars_map__ = {**parent_typevars_map, **new_typevars_map}
+                parameters = getattr(cls, '__parameters__', ())
+                parent_parameters = getattr(cls, '__pydantic_generic_metadata__', {}).get('parameters', ())
+                if parameters and parent_parameters and not all(x in parameters for x in parent_parameters):
+                    # assert Generic in bases  # tests pass with this assert, but I think we shouldn't include it
```

Comment:
Why shouldn't we include it? It seems reasonable to me, if nothing else as documentation of what this codepath expects

Reply:
Because if I am wrong and it is possible to hit this path _without_ `Generic in bases`, the AssertionError there will be a lot more confusing for a user, and have less useful context for us to debug a reported issue containing the traceback than we'd get if it hits the subsequent part of the code.

Perhaps including a listing of the base classes in the exception would be a best-of-both-worlds alternative. I'll make the change.